Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -156,14 +156,20 @@ class SourceGenerator {
     }
 
     /// returns a default build phase for a given path. This is based off the filename
-    private func getDefaultBuildPhase(for path: Path) -> TargetSource.BuildPhase? {
+    private func getDefaultBuildPhase(for path: Path, targetType: PBXProductType) -> TargetSource.BuildPhase? {
         if path.lastComponent == "Info.plist" {
             return nil
         }
         if let fileExtension = path.extension {
             switch fileExtension {
             case "swift", "m", "mm", "cpp", "c", "cc", "S", "xcdatamodeld": return .sources
             case "h", "hh", "hpp", "ipp", "tpp", "hxx", "def": return .headers
+            case "modulemap":
```

Comment:
Are there other sorts of files that would have different `CopyFilesSettings`?
If not, then we maybe don't need to pass `PBXProductType` all the way down, and these settings can be provided in `PBXProjGenerator`

Reply:
Headers for `.staticLibrary` will need to use copy files as well, so we will need to pass the `PBXProductType` down.